### PR TITLE
docs(samples): add Agent Assist code samples

### PR DESCRIPTION
**Pull request by rogers140**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-dialogflow/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [x] Ensure the tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> 🦕

### CODE REVIEW

**Comment by engelke** [comment]
This is a very big PR, well over 1000 lines of new code. It's going to be a while before I can get the time to review it.

**Comment by rogers140** [comment]
> This is a very big PR, well over 1000 lines of new code. It's going to be a while before I can get the time to review it.

Sorry about that. We are actually just migrating private code samples to public for Agent Assist Public Preview Launch. If there is a better way to do this, please let me know.

File: samples/snippets/participant_management.py [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+                                    sample_rate_herz,
+                                    stream,
+                                    timeout,
+                                    enable_extended_streaming=False):
+    """Stream audio to Dialogflow and receive transcripts and suggestions.
+
+    Args:
+        participant_name: resource name of the participant.
+        sample_rate_herz: herz rate of the sample.
+        audio_generator: a sequence of audio data.
```
Comment (engelke) [inline samples/snippets/participant_management.py:156]: This doesn't match the argument list above.

**Review by engelke — CHANGES_REQUESTED** [review]
Pass lint and all tests.
Make sure all functions have tests.
Update copyright dates if anything has been written or first published this year.
Have white space before and after import statements and internally to break code into logical chunks for readability.
Use \n in print statements only when double spacing is needed, and use consistently.

File: samples/snippets/answer_record_management.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+# limitations under the License.
+"""Dialogflow API Python sample showing how to manage AnswerRecord.
+"""
+
+
+# [START dialogflow_update_answer_record]
+def update_anwer_record(project_id, answer_record_id, click):
+    """Update the answer record.
+
+    Args:
```
[the remark this answers is not in the document]
Comment (rogers140) [reply samples/snippets/answer_record_management.py:25]: Added comments.

File: samples/snippets/conversation_management.py [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+    """Gets a specific conversation profile.
+
+    Args:
+        project_id: The GCP project linked with the conversation.
+        conversation_id: Id of the conversation."""
+    from google.cloud import dialogflow_v2 as dialogflow
+    client = dialogflow.ConversationsClient()
+    conversation_path = client.conversation_path(project_id, conversation_id)
+
+    response = client.get_conversation(name=conversation_path)
```
[the remark this answers is not in the document]
Comment (rogers140) [reply samples/snippets/conversation_management.py:62]: I checked existing code and it seems we don't quite focus on the failure handling.

File: samples/snippets/knowledge_base_management.py [hunk 4]
```diff
@@ -19,12 +19,37 @@
...
+  python knowledge_base_management.py --project-id PROJECT_ID \
+  get --knowledge-base-id knowledge_base_id
+  python knowledge_base_management.py --project-id PROJECT_ID \
+  delete --knowledge-base-id knowledge_base_id
 """
 
 import argparse
 
 
+# [START dialogflow_list_knowledge_base]
```
[the remark this answers is not in the document]
Comment (rogers140) [reply samples/snippets/knowledge_base_management.py]: Removed this function.

File: samples/snippets/analyze_content_text_test.py [hunk 5]
```diff
@@ -0,0 +1,144 @@
+# Copyright 2019 Google LLC
```
[the remark this answers is not in the document]
Comment (rogers140) [reply samples/snippets/analyze_content_text_test.py]: Fixed.

File: samples/snippets/answer_record_management.py [hunk 6]
```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env python
+
+# Copyright 2019 Google LLC
```
[the remark this answers is not in the document]
Comment (rogers140) [reply samples/snippets/answer_record_management.py]: Fixed.

File: samples/snippets/answer_record_management.py [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Dialogflow API Python sample showing how to manage AnswerRecord.
+"""
+
+
+# [START dialogflow_update_answer_record]
+def update_anwer_record(project_id, answer_record_id, click):
```
[the remark this answers is not in the document]
Comment (rogers140) [reply samples/snippets/answer_record_management.py]: Fixed.

File: samples/snippets/answer_record_management.py [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+    """Update the answer record.
+
+    Args:
+        project_id: The GCP project linked with the conversation profile.
+        answer_record_id: The answer record id returned along with the
+        suggestion."""
+    from google.cloud import dialogflow_v2 as dialogflow
```
[the remark this answers is not in the document]
Comment (rogers140) [reply samples/snippets/answer_record_management.py]: Fixed.

File: samples/snippets/participant_management.py (hunk 1 above)
Comment (rogers140) [reply samples/snippets/participant_management.py]: Fixed.

File: samples/snippets/participant_management.py [hunk 9]
```diff
@@ -0,0 +1,188 @@
...
+    participant_path = client.participant_path(project_id, conversation_id,
+                                               participant_id)
+    text_input = {'text': text, 'language_code': 'en-US'}
+    response = client.analyze_content(participant=participant_path,
+                                      text_input=text_input)
+    print('AnalyzeContent Response:')
+    print('Reply Text: {}'.format(response.reply_text))
+
+    for suggestion_result in response.human_agent_suggestion_results:
+        if suggestion_result.error is None:
```
[the remark this answers is not in the document]
Comment (rogers140) [reply samples/snippets/participant_management.py]: Fixed

**Comment by rogers140** [comment]
> Pass lint and all tests.
> Make sure all functions have tests.
> Update copyright dates if anything has been written or first published this year.
> Have white space before and after import statements and internally to break code into logical chunks for readability.
> Use \n in print statements only when double spacing is needed, and use consistently.

Fixed the issue you mentioned and better organized the tests. Removed some unnecessary functions and samples.

I think I don't know how to update the readme.rst, it seems autogenerated by readme.in but I don't know how it works. Do you know?

**Review by engelke — APPROVED** [review]

**Pull request merged**